src/Symfony/Bundle/FrameworkBundle/Secrets/SodiumVault.php

```diff
@@ -150,6 +171,8 @@ public function remove(string $name): bool
 
     public function list(bool $reveal = false): array
     {
+        $this->init();
```

Comment:
this won't have the effect you want to achieve.
instead, the mkdir should be move inside loadKeys and read-only method should be able to work when the dir is missing

Reply:
It works but it's not optimal for sure. I'll refacto the whole class then to only create the dir when it is actually needed.

Reply:
it works for you (c) but as soon as you'll try accessing a missing env var, the directory will be created - so it doesn't work for everyone :)